openeo/extra/job_management/__init__.py

```diff
@@ -619,14 +631,16 @@ def on_job_done(self, job: BatchJob, row):
 
         job_metadata = job.describe()
         job_dir = self.get_job_dir(job.job_id)
+        self.ensure_job_dir_exists(job.job_id)
         metadata_path = self.get_job_metadata_path(job.job_id)
 
-        self.ensure_job_dir_exists(job.job_id)
-        job.get_results().download_files(target=job_dir)
+        if self._download:
```

Comment:
FYI: I think this `if` can be moved to the top of this function: if you don't want to download, nothing has to happen here